source/header.c

```diff
@@ -333,6 +333,9 @@ int aws_cryptosdk_priv_hdr_parse_frame_len(
 
 int aws_cryptosdk_priv_hdr_parse_alg_suite_data(
     struct aws_cryptosdk_hdr *hdr, const struct aws_cryptosdk_alg_properties *alg_props, struct aws_byte_cursor *cur) {
+    AWS_PRECONDITION(aws_cryptosdk_hdr_is_valid(hdr));
+    AWS_PRECONDITION(&hdr->alloc != NULL);
```

Comment:
You don't need `&hdr->alloc`, right? Why not just `hdr->alloc != NULL`?

Reply:
It seems to me that the `alloc` precondition isn't necessary at all, whether @tegansb meant `&hdr->alloc` or `hdr->alloc` - the `aws_cryptosdk_hdr_is_valid(hdr)` on the previous line implies that

1. `hdr != NULL`, which in turn implies `&hdr->alloc != NULL`
2. `aws_allocator_is_valid(hdr->alloc)`, which in turn implies `hdr->alloc != NULL` according to https://github.com/awslabs/aws-c-common/blob/d0b5592e6b1efbd5fce4ee2cc8fa4aecf297e54c/source/allocator.c#L28

Reply:
You're right! I think I added this assumption before the aws_allocator_is_valid one and didn't realize I could remove it once the validity assumption was added. I've removed it now from both the harnesses and the source code.